### PR TITLE
BUG: Handle infinite values correctly in Series.argmax

### LINKED ISSUE #13595: Series.argmax() fails with np.inf
#### Code Sample, a copy-pastable example if possible

```
In [3]: s = pd.Series([1, 2, np.inf])

In [4]: s.argmax()
Out[4]: 1

In [5]: s.argmax(skipna=False)
Out[5]: nan

```
#### Expected Output

2 in both cases.
#### output of `pd.show_versions()`

In [6]: pd.show_versions()

```
INSTALLED VERSIONS
------------------
commit: a63bd12529ff309d957d714825b1753d0e02b7fa
python: 3.5.1.final.0
python-bits: 64
OS: Linux
OS-release: 4.5.0-2-amd64
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: it_IT.utf8
LOCALE: it_IT.UTF-8

pandas: 0.18.1+174.ga63bd12
nose: 1.3.7
pip: 1.5.6
setuptools: 18.4
Cython: 0.23.4
numpy: 1.10.4
scipy: 0.16.0
statsmodels: 0.8.0.dev0+111ddc0
xarray: None
IPython: 5.0.0.dev
sphinx: 1.3.1
patsy: 0.3.0-dev
dateutil: 2.2
pytz: 2012c
blosc: None
bottleneck: 1.1.0dev
tables: 3.2.2
numexpr: 2.5
matplotlib: 1.5.1
openpyxl: None
xlrd: 0.9.4
xlwt: 1.1.2
xlsxwriter: 0.7.3
lxml: None
bs4: 4.4.0
html5lib: 0.999
httplib2: 0.9.1
apiclient: 1.5.0
sqlalchemy: 1.0.11
pymysql: None
psycopg2: None
jinja2: 2.8
boto: 2.38.0
pandas_datareader: 0.2.1

```

### PR DESCRIPTION
**Pull request by matthewwillian**
- [ X] closes #13595
- [X ] tests added / passed
- [X ] passes `git diff upstream/master | flake8 --diff`
- [X ] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.18.0.txt [hunk 1]
```diff
@@ -1290,3 +1290,4 @@ Bug Fixes
 - Bug when specifying a UTC ``DatetimeIndex`` by setting ``utc=True`` in ``.to_datetime`` (:issue:`11934`)
 - Bug when increasing the buffer size of CSV reader in ``read_csv`` (:issue:`12494`)
 - Bug when setting columns of a ``DataFrame`` with duplicate column names (:issue:`12344`)
+- Bug when calling Series.argmax with infinite values (:issue: `13595`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.0.txt]: move to 0.19.0

File: pandas/tests/test_nanops.py [hunk 2]
```diff
@@ -287,6 +288,10 @@ def check_funs(self, testfunc, targfunc, allow_complex=True,
                                    allow_complex=allow_complex)
             self.check_fun(testfunc, targfunc, 'arr_obj', **kwargs)
 
+        if allow_inf:
+            self.check_fun(testfunc, targfunc, 'arr_float_inf', **kwargs)
+            self.check_fun(testfunc, targfunc, 'arr_float_neg_inf', **kwargs)
+
```
Comment (jreback) [inline pandas/tests/test_nanops.py]: also add a test that is the same as the original issue

File: doc/source/whatsnew/v0.19.0.txt [hunk 3]
```diff
@@ -549,3 +549,5 @@ Bug Fixes
 - Bug in ``groupby`` with ``as_index=False`` returns all NaN's when grouping on multiple columns including a categorical one (:issue:`13204`)
 
 - Bug where ``pd.read_gbq()`` could throw ``ImportError: No module named discovery`` as a result of a naming conflict with another python package called apiclient  (:issue:`13454`)
+
+- Bug when calling Series.argmax with infinite values (:issue: `13595`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.19.0.txt]: use double-back ticks around ``Series.argmax``

no space after ``:issue:`13595`)

**Comment by jorisvandenbossche** [comment]
@matthewwillian You have some PEP8 issues (see the travis log at the bottom: https://travis-ci.org/pydata/pandas/jobs/144182270), and a failing test.

**Comment by matthewwillian** [comment]
Okay, thank you. I will take a look.

**Comment by jorisvandenbossche** [comment]
@matthewwillian Can you update/rebase?

**Comment by sinhrks** [comment]
It's gread if you can include the fix for ``argmin`` for ``-inf``.

```
pd.Series([1, 3, -np.inf]).argmin()
0
```

**Comment by jreback** [comment]
can you rebase

**Comment by jreback** [comment]
can you rebase / update?

**Comment by jreback** [comment]
can you rebase and move release notes to 0.19.2

**Comment by jreback** [comment]
can you rebase / update

**Comment by jreback** [comment]
closing, but if you want to update, pls comment.

**Pull request closed without merging**